packages/runtime/src/helpers/functionsMetaData.ts

```diff
@@ -0,0 +1,44 @@
+import { existsSync, readJSON, writeFile } from 'fs-extra'
+import { join } from 'pathe'
+
+import { NEXT_PLUGIN, NEXT_PLUGIN_NAME } from '../constants'
+
+import { resolveModuleRoot } from './config'
+
+const checkForPackage = async (packageDir: string, nodeModule: boolean) => {
+  const packagePlugin = existsSync(packageDir) ? await readJSON(packageDir) : null
+  let nextPlugin
+  if (!nodeModule && packagePlugin) {
+    nextPlugin = packagePlugin.dependencies[NEXT_PLUGIN] ? packagePlugin.dependencies[NEXT_PLUGIN] : null
+  } else if (nodeModule && packagePlugin) {
+    nextPlugin = packagePlugin.version ? packagePlugin.version : null
+  }
+
+  return nextPlugin
+}
+
+/**
+ * Creates a function configuration file for the given function
+ *
+ * @param functionName The name of the function, e.g. `___netlify-handler`
+ * @param functionTitle The name of the function that will be displayed in logs, e.g. `Next.js SSR handler`
+ * @param functionsDir The directory where the function is located, e.g. `.netlify/functions`
+ */
+export const writeFunctionConfiguration = async (functionName: string, functionTitle: string, functionsDir: string) => {
+  const pluginPackagePath = '.netlify/plugins/package.json'
+  const ProjDir = resolveModuleRoot(NEXT_PLUGIN)
+  const nodeModulesPath = `${ProjDir}/package.json`
+
+  const nextPluginVersion =
+    (await checkForPackage(nodeModulesPath, true)) || (await checkForPackage(pluginPackagePath, false))
+
+  const metadata = {
+    config: {
+      name: functionTitle,
+      generator: `${NEXT_PLUGIN_NAME}@${nextPluginVersion || 'version-not-found'}`,
```

Comment:
It's not that the Next runtime is not found as the Next runtime is running to build the function that contains the meta data. I wonder if it makes more sense to output `'unknown'` instead. Also, I'm not even certain if we'd ever be unable to detect the version. 🤔 

```suggestion
      generator: `${NEXT_PLUGIN_NAME}@${nextPluginVersion || 'unknown'}`,
```

Reply:
I like `unknown` better than `version-not-found` as we are writing out `@netlify/next-runtime@versionHere`

Reply:
I think `unknown` is worse too. In the Humio logs we'd want to know that it's from the next runtime wouldn't we? I was thinking about something more like `next-runtime-version-not-found`. Making it clearer where the error is coming from. Thoughts?

Reply:
The generator field gets created as e.g. `@netlify/next-runtime@23.4.5` if there is a version, and in the case we can't detect the version, it'll appear as `@netlify/next-runtime@unknown`. So that's what should appear in the logs.

Reply:
Much better. Thanks, Nick!